### ISSUE TITLE
seg fault on import caffe2.python.onnx.backend 

### ISSUE THREAD

**Issue by deepali-c** [pytorch/pytorch#11074]
I am seeing a seg fault for the following snippet:
```
>>> import caffe2.python.onnx.backend as backend
Segmentation fault (core dumped)
```

The versions of caffe2, pytorch and protobuf are:
```
# packages in environment at /root/anaconda3:
#
# Name                    Version                   Build  Channel
pytorch                   0.4.1           py36_py35_py27__9.0.176_7.1.2_2    pytorch
# packages in environment at /root/anaconda3:
#
# Name                    Version                   Build  Channel
caffe2-cuda9.0-cudnn7     0.8.dev         py36_2018.08.26    caffe2
# packages in environment at /root/anaconda3:
#
# Name                    Version                   Build  Channel
libprotobuf               3.6.0                hdbcaa40_0
protobuf                  3.5.1                    py36_3    conda-forge
```

**Comment by ezyang** [comment]
cc @orionr  This is just an package version incompatibility, right?

**Comment by orionr** [comment]
Likely right. Best would be to install pytorch and caffe2 together either from source (recommended) or with one of our new binary installs. cc @pjh5 on whether those are ready yet. For source build do

```
pip uninstall torch # repeat until there is nothing left
pip uninstall caffe2 # repeat until there is nothing left
conda uninstall caffe2-cuda9.0-cudnn7
conda uninstall pytorch
git clone ...
FULL_CAFFE2=1 python setup.py install
```

**Comment by deepali-c** [comment]
I tried this, but continue to get the segfault. 

The version of `onnx `and `protobuf `is:
```
$ conda list protobuf
# packages in environment at /root/anaconda3:
#
# Name                    Version                   Build  Channel
libprotobuf               3.6.0                hdbcaa40_0
protobuf                  3.5.1                    py36_3    conda-forge
$ conda list onnx
# packages in environment at /root/anaconda3:
#
# Name                    Version                   Build  Channel
onnx                      1.0.1                    py36_1    conda-forge
```

**Comment by deepali-c** [comment]
I have built onnx first now and then built pytorch and caffe2 and observed the following error:
```
>>> import caffe2.python.onnx.backend as backend
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<path>/lib/python3.6/site-packages/caffe2/python/__init__.py", line 2, in <module>
    from caffe2.proto import caffe2_pb2
ImportError: cannot import name 'caffe2_pb2'
>>> quit()
```

**Comment by ezyang** [comment]
How did you build pytorch and caffe2? Did you build using `python setup.py develop`? This configuration is broken right now, use `python setup.py install` instead.

**Comment by ezyang** [comment]
UPDATE: Actually, it should be fixed on master. Can you pull and rebuild? Try cleaning first too.

**Comment by deepali-c** [comment]
I tried using the latest master and built using `python setup.py install` but I observe the following error:

```
>>> import caffe2.python.onnx.backend as backend
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<path>lib/python3.6/site-packages/caffe2/python/__init__.py", line 2, in <module>
    from caffe2.proto import caffe2_pb2
ImportError: cannot import name 'caffe2_pb2'
```

**Comment by deepali-c** [comment]
`find -name "caffe2_pb2*"` does not show any results.

**Comment by orionr** [comment]
Hi @deepali-c - thank you for the report! We should have a fix at https://github.com/pytorch/pytorch/pull/11517. Please verify this fixes your issue if you are able. Thanks.

**Comment by deepali-c** [comment]
Yes this works. Thanks for the fix. 
```
>>> import caffe2.python.onnx.backend as backend
>>> import numpy as np
```

**Comment by my-hello-world** [comment]
> Hi @deepali-c - thank you for the report! We should have a fix at #11517. Please verify this fixes your issue if you are able. Thanks.

same errors,but it didn't works
please
